pkg/descheduler/descheduler.go

```diff
@@ -49,6 +50,16 @@ import (
 	"sigs.k8s.io/descheduler/pkg/utils"
 )
 
+type enabledDeschedulePluginEntry struct {
+	Plugin  framework.DeschedulePlugin
+	Profile string
+}
+
+type enabledBalancePluginEntry struct {
+	Plugin  framework.BalancePlugin
+	Profile string
+}
```

Comment:
these types feel redundant, because I would expect the plugin and profile name to both be available at the time the metric is recorded.

from what I can tell, we just don't have access to the profile at plugin runtime. So if we didn't record any `profile` label, would these new structs be necessary?

Reply:
If we are not to record the metrics around profile or the metrics for strategy is not tagged along with the profile, then yes, these structs are redundant.

Reply:
I see, @knelasevero or @ingvagabund had we brought up the possibility of making profile name available to strategies?

This approach will work as it's just an implementation detail, but I'm curious if it's refactorable at some point

Reply:
From the user perspective it's better to have plugins profile unaware. A profile is a high level concept built on top of plugins. It's ok to pass a profile name (or anything else related to profiles) as long as the values are used only for monitoring, logging or decision making logic around plugins.

Reply:
@ingvagabund  Right now that is all that is being done in terms of Profiles. Their names are being used with strategy to form a unique combination that one can query from metrics as these are available as labels on them